tests/testText.cpp

```diff
@@ -118,4 +128,24 @@ BOOST_AUTO_TEST_CASE(ReadTxtAsLst)
     }
 }
 
+BOOST_AUTO_TEST_CASE(ReadOriginalFiles)
+{
+    if(!libsiedler2::test::hasS2Data)
+        return;
+    for(const auto& fileEntry : bfs::recursive_directory_iterator(libsiedler2::test::s2Path))
+    {
+        if(is_regular_file(fileEntry.status()) && fileEntry.path().extension() == ".ENG")
+        {
+            BOOST_TEST_CONTEXT("For " << fileEntry.path())
+            {
+                libsiedler2::Archiv archiv;
+                // Check only that they can be loaded and have at least 1 entry
+                BOOST_TEST_REQUIRE(libsiedler2::Load(fileEntry.path(), archiv) == 0);
```

Comment:
we could add a test here to write the read file again and compare the resulting files. The created and original files should be identical afterwards?

Reply:
That won't work for some files. Some S2 text files use the SUB byte (26 decimal) to show the EOF which is purely optional. Hence it is removed on load including everything that follows it according to the description of SUB (see wikipedia)
The save&load&compare is already done for custom created files including an example (as lst), so it wouldn't help much.

Reply:
can't we create one file specially prepared with this data to compare to?

Reply:
With which data? This loads all original files it can find just to double check that it is possible.
And as written a "prepared file" already exists: `txtAsLst.lst`